process/block/shardblock.go

```diff
@@ -258,6 +258,14 @@ func (sp *shardProcessor) ProcessBlock(
 		return err
 	}
 
+	if header.IsStartOfEpochBlock() {
+		err = sp.checkEpochCorrectnessCrossChain(chainHandler)
+		sp.epochStartTrigger.SetProcessed(header)
```

Comment:
It is ok to set processed this header even if there is an err return by checkEpochCorrectnessCrossChain ?

Reply:
changed. put the verification only if isstartofepochblock and setprocessed only if epochcorrectess does not return error.